modules/core/src/lib/attribute-transition-manager.js

```diff
@@ -204,12 +208,17 @@ export default class AttributeTransitionManager {
   // get current values of an attribute, clipped/padded to the size of the new buffer
   _getNextTransitionStates(transition) {
     const {attribute} = transition;
-    const {value, size} = attribute;
+    const {size} = attribute;
 
-    // TODO - support attribute in Transform class
-    const toState = attribute.getBuffer();
-    toState.setDataLayout({size});
+    let toState;
+    if (attribute.isGeneric) {
+      toState = {isGeneric: true, value: attribute.value, size};
+    } else {
+      toState = {isGeneric: false, buffer: attribute.getBuffer(), size};
+    }
     const fromState = transition.buffer || toState;
```

Comment:
Minor nit: Feels like some readability could be gained if these variables would be called `...Buffer` to emphasize that they contain `Buffer` instances, `initialBuffer`, `finalBuffer`